File: Solutions/Python/jump_game.py
"""
    [MEDIUM]
    55. Jump Game

    Concepts:
    - array
    - greedy

    Stats:
        Runtime | 7 ms      [Beats 97.41%]
        Memory  | 18.81 MB  [Beats 15.29%]
"""

class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        goalpost = n-1
        i = n-1

        while i >= 0:
            if goalpost - i <= nums[i]:
                goalpost = i

            i -= 1

        return goalpost == 0

File: Solutions/Cpp/jump_game.cpp
/*
    [MEDIUM]
    55. Jump Game

    Concepts:
    - array
    - greedy

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 52.30 MB  [Beats 62.30%]
*/

class Solution {
public:
    bool canJump(vector<int>& nums) {
        int goalpost = nums.size() - 1;
        int i = goalpost;

        while (i >= 0) {
            if (goalpost - i <= nums[i])
                goalpost = i;
            i--;
        }
        return (bool)(goalpost == 0);
    }
};